src/ext/peek-source.lisp

```diff
@@ -40,6 +40,7 @@
   (setf (not-switchable-buffer-p (current-buffer)) t))
 
 (define-key *peek-source-keymap* "Return" 'peek-source-select)
+(define-key *peek-source-keymap* "q" 'peek-source-quit)
```

Comment:
The peek-source is editable.
Does it malfunction when you want to insert a q?

Reply:
This is a difference between the grep interface and legit: legit's "peek-source" is a diff and is not editable.